pyface/resource/resource_manager.py

```diff
@@ -238,3 +256,80 @@ def _get_default_resource_path(self, object):
                 break
 
         return resource_path
+
+
+def _get_package_data(module, rel_path):
+    """ Return package data in bytes for the given module and resource path.
+
+    Parameters
+    ----------
+    module : ModuleType
+        A module from which package data will be discovered.
+        If the module name does not conform to the package requirement, then
+        its "__file__" attribute is used for locating the directory to search
+        for resource files.
+    rel_path : str
+        "/"-separated path for loading data file.
+
+    Returns
+    -------
+    data : bytes
+        Loaded data in bytes.
+
+    Raises
+    ------
+    OSError
+        If the path referenced does not resolve to an existing file or the
+        file cannot be read.
+    """
+
+    if (module.__spec__ is None
+            or module.__spec__.submodule_search_locations is None):
```

Comment:
This is definitely a more appropriate check than `if module.__name__ == "__main__"` as this checks the actual specific criteria of interest for handling the module specially.  
However, for my own understanding, are there any other specific cases where we expect this to be true aside from `if module.__name__ == "__main__"`?  Regardless, this is the correct thing to do, I am just curious

Reply:
I can't think of a concrete example apart from the `__main__` actually. I could make up many such modules using `importlib.util.spec_from_file_location`. There are probably some real ones out there.